### PR TITLE
MAINT: Fix CIs

### PR DESCRIPTION
**Pull request by larsoner**
In theory we should fix the failures, but let's try a quick band-aid to get CIs happy :adhesive_bandage:

### CODE REVIEW

**Review by drammock — APPROVED** [review]

File: mne/conftest.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def pytest_configure(config):
         warning_line = warning_line.strip()
         if warning_line and not warning_line.startswith("#"):
             config.addinivalue_line("filterwarnings", warning_line)
+    # TODO: Fix this with casts?
+    # https://github.com/numpy/numpy/pull/22449
+    try:
+        np.set_printoptions(legacy="1.25")
```
Comment (drammock) [inline mne/conftest.py:169]: CI failure suggests that this should actually be an integer:

```suggestion
        np.set_printoptions(legacy=125)
```

even though the example shows it as a string `"1.25"`

**Comment by scott-huberty** [comment]
Does that line in `conftest` set the array print behavior for the entire testing suite?

**Comment by larsoner** [comment]
> Does that line in conftest set the array print behavior for the entire testing suite?

Based on where I put it (`pytest_configure`) it should get called at the very beginning of the `pytest` run, before test collection even I think. `np.set_printoptions` sets a global state so it should do what we want here. Typically what we want to use is something like `with np.printoptions` context manager that *temporarily* sets a state, but in this case we want the old behavior back permanently until we adjust all our tests and such to the new `repr` type.

So in short: yes I think/hope so.

File: mne/conftest.py (hunk 1 above)
Comment (larsoner) [reply mne/conftest.py:169]: Apparently that's a newer option not compat with old versions, so I just `check_version` it instead now. That way we can use the string, which seems nicer than the int anyway. (No idea if the old versions will understand a request to use a `legacy` version that is *newer* than their version... ideally they'd ignore it but I didn't feel like finding out the hard way.)
